### PR TITLE
Update the code

### PR DESCRIPTION
**Pull request by hrayrhar**
This pull request updates the code, adds tests, adds `nnlib` as a submodule, and adds a few missing scripts.

Note that both `nnlib` and the author implementation of influence functions were published with MIT license.

### CODE REVIEW

File: sample_info/modules/influence_functions.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    for grad_elem, v_elem in zip(grads, v):
+        dot_product = dot_product + torch.sum(grad_elem * v_elem.detach())
+    return torch.autograd.grad(dot_product, params)
+
+
+@utils.with_no_grad
+def inverse_hvp_lissa(model, dataset, v, batch_size=128, scale=10,
+                      damping=0.0, num_samples=1, recursion_depth=1000):
+    """
+    reference: https://github.com/kohpangwei/influence-release/blob/master/influence/genericNeuralNet.py#L475
```
Comment (orchidmajumder) [inline sample_info/modules/influence_functions.py:69]: Is this code a `copy-paste` from this repository? What is the license of this repository?

File: sample_info/modules/influence_functions.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    """
+    model.eval()
+    inverse_hvp = None
+    loader = DataLoader(dataset=dataset, batch_size=batch_size, shuffle=True)
```
Comment (orchidmajumder) [inline sample_info/modules/influence_functions.py]: `num_workers=?`

File: sample_info/modules/ntk.py [hunk 3]
```diff
@@ -11,7 +11,16 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
+"""
+TODO: write description and notes.
```
Comment (orchidmajumder) [inline sample_info/modules/ntk.py:15]: Take care of this `TODO`?

File: sample_info/modules/stability.py [hunk 4]
```diff
@@ -11,6 +11,13 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
+"""
+TODO: write description and notes.
```
Comment (orchidmajumder) [inline sample_info/modules/stability.py]: Take care of the `TODO`.

**Review by orchidmajumder — COMMENTED** [review]
Minor comments, overall LGTM.

File: sample_info/modules/influence_functions.py (hunk 1 above)
Comment (hrayrhar) [reply sample_info/modules/influence_functions.py:69]: Almost a copy-paste. The authors' implementation had an MIT license.

File: sample_info/modules/influence_functions.py (hunk 2 above)
Comment (hrayrhar) [reply sample_info/modules/influence_functions.py]: The default value will be 0. To make it more explicit and also controllable, I have added a parameter `num_workers` to the function.

File: sample_info/modules/ntk.py (hunk 3 above)
Comment (hrayrhar) [reply sample_info/modules/ntk.py:15]: Added a single-sentence description of `ntk.py`.

File: sample_info/modules/stability.py (hunk 4 above)
Comment (hrayrhar) [reply sample_info/modules/stability.py]: Added a single-sentence description of `stability.py`.

**Comment by hrayrhar** [comment]
Hey @orchidmajumder, just a friendly reminder that this can be merged now.

Thanks

**Pull request merged**